### PR TITLE
feat(node2): add proxy support to client

### PR DESCRIPTION
**Pull request by bcluyse**
Introduces proxy support to the node2 client library by using the request library.
No new tests were written because in the way the tests are written it is not really possible unless there is an actual proxy set up.

### CODE REVIEW

**Comment by jessp01** [comment]
Hi @bcluyse,

Thank you for the pull. I understand why you didn't add a unit test (we could try to use a public proxy for it but... these tend to go up and down and cannot really be relied upon). However, please document this feature in README.md and provide a simple initialisation example.

Cheers,

**Comment by bcluyse** [comment]
Is this sufficient or should an example be added to the example.js file as well?

File: sources/node2/README.md [hunk 1]
```diff
@@ -9,6 +9,20 @@ You can install this client library using npm with:
...
 npm install kaltura-client 
 ```
+## Proxy settings for client
+If the Kaltura client has to be used behind a proxy, this can be set in the KalturaConfiguration by setting proxy
+to the url of the proxy. For example:
+
+```js
+const config = new kaltura.Configuration();
+    
+const proxyUrl = new url.URL('http://some.proxy.com');
```
Comment (jessp01) [inline sources/node2/README.md]: I believe you meant:
```js
const proxyUrl = new URL('http://some.proxy.com');
```
Right? :)

**Comment by jessp01** [comment]
The changes look good to me. Tested with and without a proxy.
@tan-tan-kanarek - any comments from your end? If not, we can merge to 15.5.0, to be created on Monday.

**Comment by tan-tan-kanarek** [comment]
LGTM
@irena-l, could you please make sure it doesn't affect your changes?

**Comment by jessp01** [comment]
@irena-l - any feedback? I'd like to merge now that 15.5.0 was created.

**Comment by tan-tan-kanarek** [comment]
Breaks agent options: 
https://github.com/kaltura/clients-generator/blob/Orion-15.11.0/sources/node2/KalturaClientBase.js#L173

**Comment by jessp01** [comment]
Hi @bcluyse,

Please see @tan-tan-kanarek's comment above. The original code read:
```
	var httpInterface = options.protocol === 'http:' ? http : https;
		if (client.config.agentOptions) {
			if (!client.agent) {
				client.agent = new httpInterface.Agent(client.config.agentOptions);
			}
			options.agent = client.agent;
		}
```

Please submit a pull to correct this.

Thanks,

**Comment by bcluyse** [comment]
Hi,

Is there any more information available as to what exactly is broken?
I will then fix it asap.

Kind regards

**Comment by tan-tan-kanarek** [comment]
Already fixed by @guykatznelsonkaltura: https://github.com/kaltura/clients-generator/commit/8a27984cdfd7957cfa6968529cd6e471dde41247
I asked him to revert it and commit to new branch for proper working procedures.

**Pull request merged**